Test_runDispatcher/no_gateway_actions,_dispatcher_is_flushed is flaky on Windows 10

From the most recent flaky tests report

```
extended windows / Matrix - PLATFORM = 'windows-10' / Test / Test_runDispatcher – github.com/elastic/elastic-agent/internal/pkg/agent/application	failed 1 times:
 	elastic-agent	failed 1 times
extended windows / Matrix - PLATFORM = 'windows-10' / Test / Test_runDispatcher/no_gateway_actions,_dispatcher_is_flushed – github.com/elastic/elastic-agent/internal/pkg/agent/application	failed 1 times:
 	elastic-agent	failed 1 times
```
## Flaky Test

* **Test Name:** Test_runDispatcher/no_gateway_actions,_dispatcher_is_flushed 
* **Link:** https://github.com/elastic/elastic-agent/blob/7506b3d89477c8643845c632b3820a9b55ca0e27/internal/pkg/agent/application/managed_mode_test.go#L103
* **Branch:** main, 8.8, 8.7

### Stack Trace

```
[2023-04-24T20:48:45.286Z] === Failed
[2023-04-24T20:48:45.286Z] === FAIL: internal/pkg/agent/application Test_runDispatcher/no_gateway_actions,_dispatcher_is_flushed (0.13s)
[2023-04-24T20:48:45.286Z]     --- FAIL: Test_runDispatcher/no_gateway_actions,_dispatcher_is_flushed (0.13s)
[2023-04-24T20:48:45.286Z] panic: 
[2023-04-24T20:48:45.286Z] assert: mock: The method has been called over 1 times.
[2023-04-24T20:48:45.286Z] 	Either do one more Mock.On("Dispatch").Return(...), or remove extra call.
[2023-04-24T20:48:45.286Z] 	This call was unexpected:
[2023-04-24T20:48:45.286Z] 		Dispatch(*context.timerCtx,*application.mockAcker,[]fleetapi.Action)
[2023-04-24T20:48:45.286Z] 		0: &context.timerCtx{cancelCtx:context.cancelCtx{Context:(*context.emptyCtx)(0xc000038128), mu:sync.Mutex{state:0, sema:0x0}, done:atomic.Value{v:(chan struct {})(0xc00057c5a0)}, children:map[context.canceler]struct {}(nil), err:error(nil)}, timer:(*time.Timer)(0xc000538320), deadline:time.Time{wall:0xc109d8906bdb37f8, ext:377343601, loc:(*time.Location)(0x1423680)}}
[2023-04-24T20:48:45.286Z] 		1: &application.mockAcker{Mock:mock.Mock{ExpectedCalls:[]*mock.Call(nil), Calls:[]mock.Call(nil), test:mock.TestingT(nil), testData:objx.Map(nil), mutex:sync.Mutex{state:0, sema:0x0}}}
[2023-04-24T20:48:45.286Z] 		2: []fleetapi.Action(nil)
[2023-04-24T20:48:45.286Z] 	at: [C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-2545/src/github.com/elastic/elastic-agent/internal/pkg/agent/application/managed_mode_test.go:25 C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-2545/src/github.com/elastic/elastic-agent/internal/pkg/agent/application/managed_mode.go:229 C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-2545/src/github.com/elastic/elastic-agent/internal/pkg/agent/application/managed_mode_test.go:126] [recovered]
[2023-04-24T20:48:45.286Z] 	panic: 
[2023-04-24T20:48:45.286Z] assert: mock: The method has been called over 1 times.
[2023-04-24T20:48:45.286Z] 	Either do one more Mock.On("Dispatch").Return(...), or remove extra call.
[2023-04-24T20:48:45.286Z] 	This call was unexpected:
[2023-04-24T20:48:45.286Z] 		Dispatch(*context.timerCtx,*application.mockAcker,[]fleetapi.Action)
[2023-04-24T20:48:45.286Z] 		0: &context.timerCtx{cancelCtx:context.cancelCtx{Context:(*context.emptyCtx)(0xc000038128), mu:sync.Mutex{state:0, sema:0x0}, done:atomic.Value{v:(chan struct {})(0xc00057c5a0)}, children:map[context.canceler]struct {}(nil), err:error(nil)}, timer:(*time.Timer)(0xc000538320), deadline:time.Time{wall:0xc109d8906bdb37f8, ext:377343601, loc:(*time.Location)(0x1423680)}}
[2023-04-24T20:48:45.286Z] 		1: &application.mockAcker{Mock:mock.Mock{ExpectedCalls:[]*mock.Call(nil), Calls:[]mock.Call(nil), test:mock.TestingT(nil), testData:objx.Map(nil), mutex:sync.Mutex{state:0, sema:0x0}}}
[2023-04-24T20:48:45.286Z] 		2: []fleetapi.Action(nil)
[2023-04-24T20:48:45.286Z] 	at: [C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-2545/src/github.com/elastic/elastic-agent/internal/pkg/agent/application/managed_mode_test.go:25 C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-2545/src/github.com/elastic/elastic-agent/internal/pkg/agent/application/managed_mode.go:229 C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-2545/src/github.com/elastic/elastic-agent/internal/pkg/agent/application/managed_mode_test.go:126]
[2023-04-24T20:48:45.286Z] 
[2023-04-24T20:48:45.286Z] goroutine 79 [running]:
[2023-04-24T20:48:45.286Z] testing.tRunner.func1.2({0xb8a5c0, 0xc0001cf530})
[2023-04-24T20:48:45.286Z] 	C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-2545/.gvm/versions/go1.19.8.windows.amd64/src/testing/testing.go:1396 +0x24e
[2023-04-24T20:48:45.286Z] testing.tRunner.func1()
[2023-04-24T20:48:45.286Z] 	C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-2545/.gvm/versions/go1.19.8.windows.amd64/src/testing/testing.go:1399 +0x39f
[2023-04-24T20:48:45.286Z] panic({0xb8a5c0, 0xc0001cf530})
[2023-04-24T20:48:45.286Z] 	C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-2545/.gvm/versions/go1.19.8.windows.amd64/src/runtime/panic.go:884 +0x212
[2023-04-24T20:48:45.286Z] github.com/stretchr/testify/mock.(*Mock).fail(0xc000696870, {0xd11e39?, 0x4?}, {0xc0002be300?, 0x3?, 0x3?})
[2023-04-24T20:48:45.287Z] 	C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-2545/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:328 +0x145
[2023-04-24T20:48:45.287Z] github.com/stretchr/testify/mock.(*Mock).MethodCalled(0xc000696870, {0xfa7f89, 0x8}, {0xc00052e030, 0x3, 0x3})
[2023-04-24T20:48:45.287Z] 	C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-2545/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:472 +0x2ac
[2023-04-24T20:48:45.287Z] github.com/stretchr/testify/mock.(*Mock).Called(0x0?, {0xc00052e030, 0x3, 0x3})
[2023-04-24T20:48:45.287Z] 	C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-2545/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:456 +0x148
[2023-04-24T20:48:45.287Z] github.com/elastic/elastic-agent/internal/pkg/agent/application.(*mockDispatcher).Dispatch(0xc000696870, {0xe33448?, 0xc000095380}, {0xe2ffb8?, 0xc0005382d0}, {0x0, 0x0, 0x0})
[2023-04-24T20:48:45.287Z] 	C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-2545/src/github.com/elastic/elastic-agent/internal/pkg/agent/application/managed_mode_test.go:25 +0x125
[2023-04-24T20:48:45.287Z] github.com/elastic/elastic-agent/internal/pkg/agent/application.runDispatcher({0xe33448, 0xc000095380}, {0xe2ffe0, 0xc000696870}, {0xe33640, 0xc000696820}, {0xe2ffb8, 0xc0005382d0}, 0x121ee90?)
[2023-04-24T20:48:45.287Z] 	C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-2545/src/github.com/elastic/elastic-agent/internal/pkg/agent/application/managed_mode.go:229 +0x17f
[2023-04-24T20:48:45.287Z] github.com/elastic/elastic-agent/internal/pkg/agent/application.Test_runDispatcher.func7(0xc000722680?)
[2023-04-24T20:48:45.287Z] 	C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-2545/src/github.com/elastic/elastic-agent/internal/pkg/agent/application/managed_mode_test.go:126 +0xec
[2023-04-24T20:48:45.287Z] testing.tRunner(0xc000722ea0, 0xc0007177e0)
[2023-04-24T20:48:45.287Z] 	C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-2545/.gvm/versions/go1.19.8.windows.amd64/src/testing/testing.go:1446 +0x10b
[2023-04-24T20:48:45.287Z] created by testing.(*T).Run
[2023-04-24T20:48:45.287Z] 	C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-2545/.gvm/versions/go1.19.8.windows.amd64/src/testing/testing.go:1493 +0x35f
[2023-04-24T20:48:45.287Z] 
[2023-04-24T20:48:45.287Z] === FAIL: internal/pkg/agent/application Test_runDispatcher (0.34s)
[2023-04-24T20:48:45.287Z] 
```

--- a/internal/pkg/agent/application/managed_mode_test.go
+++ b/internal/pkg/agent/application/managed_mode_test.go
@@ -102,8 +102,7 @@ func Test_runDispatcher(t *testing.T) {
 		},
 		interval: time.Second,
 	}, {
-		name:                "no gateway actions, dispatcher is flushed",
-		skipOnWindowsReason: "Flaky test: https://github.com/elastic/elastic-agent/issues/2585",
+		name: "no gateway actions, dispatcher is flushed",
 		mockGateway: func(ch chan []fleetapi.Action) *mockGateway {
 			gateway := &mockGateway{}
 			gateway.On("Actions").Return((<-chan []fleetapi.Action)(ch))
@@ -112,6 +111,7 @@ func Test_runDispatcher(t *testing.T) {
 		mockDispatcher: func() *mockDispatcher {
 			dispatcher := &mockDispatcher{}
 			dispatcher.On("Dispatch", mock.Anything, mock.Anything, mock.Anything).Once()
+			dispatcher.On("Dispatch", mock.Anything, mock.Anything, mock.Anything).Maybe() // allow a second call in case there are timing issues in the CI pipeline
 			return dispatcher
 		},
 		interval: time.Millisecond * 60,

